### PR TITLE
Import saf via URL

### PR DESCRIPTION
**Pull request by enea4science**
## Description
This task adds the Import saf via url needed for: this https://github.com/DSpace/DSpace/pull/8710 

Closes https://github.com/DSpace/dspace-angular/issues/1913

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by enea4science** [comment]
Hello @artlowel 
It was a misunderstanding from my part when opening the PR, I edited the title and the description.

File: src/app/item-page/orcid-page/orcid-sync-settings/orcid-sync-settings.component.ts [hunk 1]
```diff
@@ -156,7 +156,8 @@ export class OrcidSyncSettingsComponent implements OnInit {
         }
       }),
     ).subscribe((remoteData: RemoteData<ResearcherProfile>) => {
-      if (remoteData.isSuccess) {
+      // hasSucceeded is true if the response is success or successStale
+      if (remoteData.hasSucceeded) {
```
Comment (tdonohue) [inline src/app/item-page/orcid-page/orcid-sync-settings/orcid-sync-settings.component.ts:160]: This change seems unrelated to this PR?

**Review by tdonohue — COMMENTED** [review]
@enea4science : I tested this today, and I cannot get the Import from URL to work.

1. Login as an Admin
2. Go to "Import -> Batch Import (Zip)"
3. Select a Collection
4. Deselect "validate only"
5. Change Toggle to "URL"
6. Paste in a URL of a SAF Zip file.  I used this one which works if I import via drag and drop: https://github.com/DSpace-Labs/AIP-Files/releases/download/demo-entities-data/SAF-Import-Entities.zip
7. Click Proceed.

After a few seconds, I see this error in the output:
```
2023-06-06 21:51:58.549 INFO import - 54 @ The script has started
2023-06-06 21:51:58.551 INFO import - 54 @ Destination collections:
2023-06-06 21:51:58.551 INFO import - 54 @ Owning Collection: Test Import from URL Collection
2023-06-06 21:51:59.075 INFO import - 54 @ Started: 1686088318549
2023-06-06 21:51:59.075 INFO import - 54 @ Ended: 1686088319074
2023-06-06 21:51:59.075 INFO import - 54 @ Elapsed time: 0 secs (525 msecs)
2023-06-06 21:51:59.075 ERROR import - 54 @ null
2023-06-06 21:51:59.075 ERROR import - 54 @ java.lang.Exception: Error committing changes to database: Authorization attempted on null DSpace object READ by user 335647b6-8a52-4ecb-a8c1-7ebabb199bda, aborting most recent changes
	at org.dspace.app.itemimport.ItemImport.internalRun(ItemImport.java:233)
	at org.dspace.scripts.DSpaceRunnable.run(DSpaceRunnable.java:104)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.dspace.authorize.AuthorizeException: Authorization attempted on null DSpace object READ by user 335647b6-8a52-4ecb-a8c1-7ebabb199bda
	at org.dspace.authorize.AuthorizeServiceImpl.authorizeAction(AuthorizeServiceImpl.java:144)
	at org.dspace.authorize.AuthorizeServiceImpl.authorizeAction(AuthorizeServiceImpl.java:120)
	at org.dspace.authorize.AuthorizeServiceImpl.authorizeAction(AuthorizeServiceImpl.java:114)
	at org.dspace.content.BitstreamServiceImpl.retrieve(BitstreamServiceImpl.java:298)
	at org.dspace.app.rest.scripts.handler.impl.RestDSpaceRunnableHandler.getFileStream(RestDSpaceRunnableHandler.java:237)
	at org.dspace.app.itemimport.ItemImport.readZip(ItemImport.java:345)
	at org.dspace.app.itemimport.ItemImport.internalRun(ItemImport.java:224)
	... 4 more
```

That error almost looks like the feature isn't working anymore that @francescopioscognamiglio built in #8710 (which is already merged).  But, I'm not sure what is causing this problem because I thought this feature worked previously.

If you or @francescopioscognamiglio have any ideas, I'd appreciate it.  I cannot seem to get this to work.

In the meantime, I have a small inline comment below.

**Comment by enea4science** [comment]
Hello @tdonohue 
I tried from the processes UI and I get the same error. 

image

From what I understand the feature should work basically if was done from here: 

image

**Comment by francescopioscognamiglio** [comment]
Hi @tdonohue, you are right that it was working fine in https://github.com/DSpace/DSpace/pull/8710.
I did some analysis and found that something was changed after our testing https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/app/itemimport/ItemImport.java#L345 and now the import via URL no longer works.
I will open a bug on the backend side and try to do a quick fix about that.

File: src/app/item-page/orcid-page/orcid-sync-settings/orcid-sync-settings.component.ts (hunk 1 above)
Comment (tdonohue) [reply src/app/item-page/orcid-page/orcid-sync-settings/orcid-sync-settings.component.ts:160]: I think this should be removed or explained? It doesn't seem to have anything to do with Importing SAF files.

**Review by tdonohue — APPROVED** [review]
👍 @enea4science : Re-tested today with the bug fix in https://github.com/DSpace/DSpace/pull/8894.  It works!  However, before I can merge it, I think we need to remove the code unrelated to this PR.  See inline below.
